src/mame/video/segasaturn_vdp1.h

```diff
@@ -0,0 +1,286 @@
+// license:LGPL-2.1+
+// copyright-holders:David Haywood, Angelo Salese, Olivier Galibert, Mariusz Wojcieszek, R. Belmont
+/**************************************************************************************************
+
+Sega Saturn / ST-V - VDP1
+
+**************************************************************************************************/
+
+#ifndef MAME_VIDEO_SATURN_VDP1_H
+#define MAME_VIDEO_SATURN_VDP1_H
+
+#pragma once
+
+
+#define RGB_R(_color)   (_color & 0x1f)
+#define RGB_G(_color)   ((_color >> 5) & 0x1f)
+#define RGB_B(_color)   ((_color >> 10) & 0x1f)
+
+#define SWAP_INT32(_a,_b) \
+	{ \
+		int32_t t; \
+		t = _a; \
+		_a = _b; \
+		_b = t; \
+	}
+
+#define SWAP_INT32PTR(_p1, _p2) \
+	{ \
+		int32_t *p; \
+		p = _p1; \
+		_p1 = _p2; \
+		_p2 = p; \
+	}
+
+/*TV Mode Selection Register */
+/*
+   xxxx xxxx xxxx ---- | UNUSED
+   ---- ---- ---- x--- | VBlank Erase/Write (VBE)
+   ---- ---- ---- -xxx | TV Mode (TVM)
+   TV-Mode:
+   This sets the Frame Buffer size,the rotation of the Frame Buffer & the bit width.
+   bit 2 HDTV disable(0)/enable(1)
+   bit 1 non-rotation/rotation(1)
+   bit 0 16(0)/8(1) bits per pixel
+   Size of the Frame Buffer:
+   7 invalid
+   6 invalid
+   5 invalid
+   4 512x256
+   3 512x512
+   2 512x256
+   1 1024x256
+   0 512x256
+*/
+#define STV_VDP1_TVMR ((m_vdp1_regs[0x000/2])&0xffff)
+#define STV_VDP1_VBE  ((STV_VDP1_TVMR & 0x0008) >> 3)
+#define STV_VDP1_TVM  ((STV_VDP1_TVMR & 0x0007) >> 0)
```

Comment:
These and most of the rest of the macros for accessing members really shouldn’t be leaking downstream from the header.  Can they be turned into const-qualified inline member functions, or failing that, moved to the .cpp file?  Making them into member functions also makes it easier to work with in debuggers.